galaxy/api/v2/views/collection.py

```diff
@@ -76,6 +76,7 @@ def post(self, request, *args, **kwargs):
                 'repository_pk': repository.pk,
                 'namespace_pk': namespace.pk,
                 'task_id': import_task.id,
+                'filename': data['filename'],
```

Comment:
@cutwater I didn't see a way to access filename from artifact object, so passed it forward here

Reply:
It's ok